### PR TITLE
Release notes v0.36 — error estimation section

### PR DESCRIPTION
**Pull request by DSGuala**
**Context:**
We want to communicate the new error estimation functionality in v0.36 to the user.

**Description of the Change:**

Adds a section in the release notes describing the change and sharing code examples.

### CODE REVIEW

File: doc/releases/changelog-0.36.0.md [hunk 1]
```diff
@@ -6,28 +6,64 @@
...
-* Added `error` method to `QuantumPhaseEstimation` template.
-  [(#5278)](https://github.com/PennyLaneAI/pennylane/pull/5278)
+* This version of PennyLane lays the foundation for error estimation. [(#5154)](https://github.com/PennyLaneAI/pennylane/pull/5154)[(#5464)](https://github.com/PennyLaneAI/pennylane/pull/5464)[(#5465)](https://github.com/PennyLaneAI/pennylane/pull/5465)
 
-* Added new `SpectralNormError` class to the new error tracking functionality.
-  [(#5154)](https://github.com/PennyLaneAI/pennylane/pull/5154)
-
-* The `qml.TrotterProduct` operator now supports error estimation functionality. 
-  [(#5384)](https://github.com/PennyLaneAI/pennylane/pull/5384)
+  * For example, track the spectral norm error of custom operations by using the new `SpectralNormError` class and the error tracking functionality in `qml.specs` and `qml.Tracker`.
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:11]:
```suggestion
  * For algorithmic-level usecases it is useful to specify a custom operation representing a
    major building block of the algorithm. This building block can have an error attached but
    does not require a decomposition or matrix representation:
```

File: doc/releases/changelog-0.36.0.md [hunk 2]
```diff
@@ -6,28 +6,64 @@
...
+      def compute_matrix(*params, **hyperparams):
+          return np.array([[0,1],[1,0]])
+      
+  dev = qml.device('default.qubit', wires=2)
+  @qml.qnode(dev)
+  def circuit():
+      XWithError(0.1,[0])
+      XWithError(0.2,[1])
+      return qml.state()
+  ```
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:34]:
````suggestion
    ```python
    import pennylane as qml
    from pennylane.resource.error import ErrorOperation, SpectralNormError

    class XWithError(ErrorOperation):
        def __init__(self, error_val, wires):
            self.error_val = error_val
            super().__init__(wires=wires)
        def error(self):
            return SpectralNormError(self.error_val)

    dev = qml.device("default.qubit")

    @qml.qnode(dev)
    def circuit():
        XWithError(0.1,[0])
        XWithError(0.2,[1])
        return qml.state()
    ```
````
Note that devices generally no longer need to have the number of wires specified.

File: doc/releases/changelog-0.36.0.md [hunk 3]
```diff
@@ -6,28 +6,64 @@
...
+  Using `qml.specs` on this circuit returns the additive spectral norm error:
+  ```pycon
+  >>> qml.specs(circuit)()['errors']
+  {'SpectralNormError': SpectralNormError(0.30000000000000004)}
+  ```
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:39]:
````suggestion

    The total spectral norm error of the circuit can be calculated using `qml.specs`:

    ```pycon
    >>> qml.specs(circuit)()['errors']
    {'SpectralNormError': SpectralNormError(0.30000000000000004)}
    ```
````

File: doc/releases/changelog-0.36.0.md [hunk 4]
```diff
@@ -6,28 +6,64 @@
...
+  Alternatively, use `qml.Tracker` to obtain these results:
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:40]:
```suggestion
```

File: doc/releases/changelog-0.36.0.md [hunk 5]
```diff
@@ -6,28 +6,64 @@
...
   ```pycon
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:41]:
```suggestion
```

File: doc/releases/changelog-0.36.0.md [hunk 6]
```diff
@@ -6,28 +6,64 @@
...
-  >>> op = qml.TrotterProduct(hamiltonian, time=0.01, order=2)
-  >>> op.error(method="one-norm")
-  SpectralNormError(8.039062500000003e-06)
-  >>>
-  >>> op.error(method="commutator")
-  SpectralNormError(6.166666666666668e-06)
+  >>> with qml.Tracker(dev) as tracker:
+  ...     circuit()
+  >>> tracker['history']['errors']
+  [{'SpectralNormError': SpectralNormError(0.30000000000000004)}]
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:45]:
```suggestion
```

File: doc/releases/changelog-0.36.0.md [hunk 7]
```diff
@@ -6,28 +6,64 @@
...
   ```
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:46]:
```suggestion
```

File: doc/releases/changelog-0.36.0.md (hunk 4 above)
Comment (trbromley) [reply doc/releases/changelog-0.36.0.md:40]: We can skip the tracker mention here unless we think it's useful?

File: doc/releases/changelog-0.36.0.md [hunk 8]
```diff
@@ -6,28 +6,64 @@
...
 
-* `qml.specs` and `qml.Tracker` now return information about algorithmic errors for the qnode as well.
-  [(#5464)](https://github.com/PennyLaneAI/pennylane/pull/5464)
-  [(#5465)](https://github.com/PennyLaneAI/pennylane/pull/5465)
+  * Also track the spectral norm error of supported operations: `QuantumPhaseEstimation` and `TrotterProduct` [(#5278)](https://github.com/PennyLaneAI/pennylane/pull/5278)[(#5384)](https://github.com/PennyLaneAI/pennylane/pull/5384)
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:48]:
```suggestion
  * PennyLane already includes a number of built-in building blocks like
    `QuantumPhaseEstimation` and `TrotterProduct`, both of which now
    propagate errors that depend upon their input arguments such as
    the number of steps performed in the Trotter product:
    [(#5278)](https://github.com/PennyLaneAI/pennylane/pull/5278)
    [(#5384)](https://github.com/PennyLaneAI/pennylane/pull/5384)
```

File: doc/releases/changelog-0.36.0.md [hunk 9]
```diff
@@ -6,28 +6,64 @@
...
+  def circuit():
+    qml.TrotterProduct(hamiltonian, time=0.5, order=2)
+    qml.QuantumPhaseEstimation(XWithError(0.1,wires=0),estimation_wires=[1,2,3])
+    return qml.state()
+  ```
+
+  ```pycon
+  >>> qml.specs(circuit)()['history']
+  {'SpectralNormError': SpectralNormError(1.4708333333333332)}
+  ```
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:66]:
````suggestion
    ```python
    import pennylane as qml
    
    dev = qml.device('default.qubit')
    hamiltonian = qml.dot([1.0, 0.5, -0.25], [qml.X(0), qml.Y(0), qml.Z(0)])

    @qml.qnode(dev)
    def circuit():
        qml.TrotterProduct(hamiltonian, time=0.1, order=2)
        qml.QuantumPhaseEstimation(XWithError(0.01, wires=0), estimation_wires=[1, 2, 3])
        return qml.state()
    ```

    ```pycon
    >>> qml.specs(circuit)()["errors"]
    {'SpectralNormError': SpectralNormError(0.07616666666666666)}
    ```
````

File: doc/releases/changelog-0.36.0.md [hunk 10]
```diff
@@ -6,28 +6,64 @@
...
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:67]:
```suggestion

  Error estimation can also be combined with resource estimation, allowing you
  to investigate the tradeoff between resources and errors.

```
@DSGuala I'm wondering if we should lean on this even more. We've focused on this part when creating our release illustration.

**Review by trbromley — CHANGES_REQUESTED** [review]
Thanks @DSGuala! I've left some suggestions.

File: doc/releases/changelog-0.36.0.md (hunk 4 above)
Comment (DSGuala) [reply doc/releases/changelog-0.36.0.md:40]: Not mandatory, committing the suggestion :+1:

File: doc/releases/changelog-0.36.0.md (hunk 10 above)
Comment (DSGuala) [reply doc/releases/changelog-0.36.0.md:67]: Yup, I can add a brief example of trading off resources vs errors here.

File: doc/releases/changelog-0.36.0.md [hunk 11]
```diff
@@ -6,28 +6,74 @@
...
-* Added new `SpectralNormError` class to the new error tracking functionality.
+* This version of PennyLane lays the foundation for estimating the total error
+  in a quantum circuit from the combination of individual gate errors.
   [(#5154)](https://github.com/PennyLaneAI/pennylane/pull/5154)
+  [(#5464)](https://github.com/PennyLaneAI/pennylane/pull/5464)
+  [(#5465)](https://github.com/PennyLaneAI/pennylane/pull/5465)
 
-* The `qml.TrotterProduct` operator now supports error estimation functionality. 
-  [(#5384)](https://github.com/PennyLaneAI/pennylane/pull/5384)
+  Focusing on the [spectral norm](https://en.wikipedia.org/wiki/Matrix_norm#Spectral_norm_(p_=_2))
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:15]:
```suggestion
  Focusing on the [spectral norm](https://en.wikipedia.org/wiki/Matrix_norm)
```

File: doc/releases/changelog-0.36.0.md [hunk 12]
```diff
@@ -6,28 +6,74 @@
...
-  ```pycon
-  >>> hamiltonian = qml.dot([1.0, 0.5, -0.25], [qml.X(0), qml.Y(0), qml.Z(0)])
-  >>> op = qml.TrotterProduct(hamiltonian, time=0.01, order=2)
-  >>> op.error(method="one-norm")
-  SpectralNormError(8.039062500000003e-06)
-  >>>
-  >>> op.error(method="commutator")
-  SpectralNormError(6.166666666666668e-06)
-  ```
+  * For algorithmic-level usecases it is useful to specify a custom operation representing a
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:18]:
```suggestion
  * For algorithmic-level usecases it is useful to create a custom operation representing a
```

**Review by trbromley — APPROVED** [review]
Thanks @DSGuala!

**Pull request merged**